### PR TITLE
Full Unit Test Migration from old framework

### PR DESCRIPTION
**Pull request by pkanher617**
This PR finalizes the Unit Test migration started in PR #645. 

All unit tests have been converted in to mocha/chai framework assertion language, and are in passing state. Minor changes to UserAgentApplication were made to accommodate some of these changes. Namely:
- removal of urlContainsQueryStringParameter, since it is no longer used or needed
- a getter for the current msal configuration, since there is no current way to view the msal config state

Test files for UserAgentApplication and ServerRequestParameters have been added and reflect all tests that existed in MSAL.spec.ts and MSAL.spec.js

### CODE REVIEW

File: lib/msal-core/src/UserAgentApplication.ts [hunk 1]
```diff
@@ -2211,21 +2210,20 @@ export class UserAgentApplication {
...
    */
-  private urlContainsQueryStringParameter(name: string, url: string): boolean {
-    // regex to detect pattern of a ? or & followed by the name parameter and an equals character
-    const regex = new RegExp("[\\?&]" + name + "=");
-    return regex.test(url);
+  public getCurrentConfiguration(): Configuration {
+    if (!this.config) {
+      throw ClientConfigurationError.createNoSetConfigurationError();
+    }
+    return this.config;
```
Comment (DarylThayil) [inline lib/msal-core/src/UserAgentApplication.ts]: what does this enable?

File: lib/msal-core/src/error/ClientConfigurationError.ts [hunk 2]
```diff
@@ -5,6 +5,10 @@ import { Constants } from "../Constants";
 import { ClientAuthError } from "./ClientAuthError";
 
 export const ClientConfigurationErrorMessage = {
+    configurationNotSet: {
```
Comment (DarylThayil) [inline lib/msal-core/src/error/ClientConfigurationError.ts]: is this possible? doesnt UAA throw without this

File: lib/msal-core/src/UserAgentApplication.ts (hunk 1 above)
Comment (pkanher617) [reply lib/msal-core/src/UserAgentApplication.ts]: Enables us to see the state of the config of the msal object.

File: lib/msal-core/src/error/ClientConfigurationError.ts (hunk 2 above)
Comment (pkanher617) [reply lib/msal-core/src/error/ClientConfigurationError.ts]: More of a sanity check error than anything else.

File: lib/msal-core/test/ServerRequestParameters.spec.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            let authenticationRequestParameters: ServerRequestParameters;
+            let authority: Authority;
+            authority = AuthorityFactory.CreateInstance("https://login.microsoftonline.com/common/", this.validateAuthority);
+            authenticationRequestParameters = new ServerRequestParameters(authority, "0813e1d1-ad72-46a9-8665-399bba48c201", ["user.read"], "id_token", "", "https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-implicit-grant-flow?name=value&name2=value2");
+            var result;
+            result = authenticationRequestParameters.createNavigationUrlString(["user.read"]);
+            expect(decodeURIComponent(result[4])).to.include("https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-implicit-grant-flow?name=value&name2=value2");
+        });
+    });
+});
```
[the remark this answers is not in the document]
Comment (pkanher617) [reply lib/msal-core/test/ServerRequestParameters.spec.ts]: Yes that is coming in a future PR

File: lib/msal-core/src/UserAgentApplication.ts [hunk 4]
```diff
@@ -2211,21 +2210,20 @@ export class UserAgentApplication {
...
   }
```
[the remark this answers is not in the document]
Comment (pkanher617) [reply lib/msal-core/src/UserAgentApplication.ts]: Possibly, but I will review when I submit the PR for those Unit Tests

**Pull request merged**